llvm/lib/Target/AIE/AIE2InstructionSelector.cpp

```diff
@@ -889,6 +897,31 @@ bool AIE2InstructionSelector::selectG_CONCAT_VECTORS(MachineInstr &I,
   return true;
 }
 
+// Try to match BRCOND(Intrinsic::loop_decrement)
+bool AIE2InstructionSelector::selectBrCondLoopDecrement(
+    MachineInstr &BrCond, MachineRegisterInfo &MRI) {
+  MachineOperand &MO = BrCond.getOperand(0);
```

Comment:
Super-nit: Maybe have an assert to verify the `BrCond.getOpcode() == G_BRCOND` precoindition?